### PR TITLE
Integrated valence

### PR DESCRIPTION
**Pull request by campd**
In case anyone is interested, here's a quick hack-n-slash job to do the valence work in-page.

This doesn't obviate the need for a proxy, because a) we still need it for firefox and b) we can't access chrome's tab list from the page (same origin policy) so we need to proxy that.  But you should be able to debug chrome from chrome without Firefox running.

### CODE REVIEW

File: client/framework/toolbox.js [hunk 1]
```diff
@@ -85,8 +85,18 @@ const ToolboxButtons = exports.ToolboxButtons = [
...
 exports.getWSTarget = function() {
-  let socket = new WebSocket("ws://localhost:9000");
-  let transport = new DebuggerTransport(socket);
+  let transport;
+  if (false) {
+    // Connect to chromium.  Need to fetch the tab list since it's over
+    // http and not same-origin, but will connect directly to the web
+    // socket.
+    let Valence = require("devtools/valence/lib/chromium/server");
+    transport = Valence.connect("http://localhost:8081/chrome-tab-list");
```
Comment (jlongster) [inline client/framework/toolbox.js]: this needs to point to `/chrome-tab-list/json` I think.

**Comment by jlongster** [comment]
I tried this and couldn't get it to work. I see the following:

screen shot 2015-12-02 at 8 41 08 pm

I may have set things up wrongly. I ran `gulp start`, started Chrome with a debugging port of 9222 and then tried to open the toolbox in Firefox. I also changed the `getWSTarget` function to use valence.

**Comment by jlongster** [comment]
(This would be hugely helpful debugging problems, now I can just add console.log's in that code. would be super nice)

File: client/framework/toolbox.js (hunk 1 above)
Comment (jlongster) [inline client/framework/toolbox.js]: Oh I was wrong. I see the URL is it connecting to is the correct one.

**Comment by jlongster** [comment]
Just noticed this:

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8081/chrome-tab-list/json

Just realized I have to run it through our local server now, not my static fileserver. I'll see if that works.

**Comment by jlongster** [comment]
Ok got it working going to the right URL.

This seems to break the debugger. Here `DebuggerServer` is not defined: https://github.com/joewalker/devtools.html/blob/master/server/actors/string.js#L22 See it's commented out at the top, because that file does not exist here. I haven't tried the inspector.

**Comment by campd** [comment]
You know, I don't pull request much :/

Fixed the string problem, rebased the commits.  New pull should fix it.  Let me know how it goes.

**Comment by jlongster** [comment]
This works great for me, and removes some of the abstraction which is nice. It's easier to figure out what's going on. I was able to get node working by manually tweaking the valence code which was way harder than before (not going to commit that).

I've been testing that and it works. I'm going to merge it but we can revert it tomorrow if anyone has problems.

**Pull request merged**